### PR TITLE
[Repository Manager] Continue repo downloading

### PR DESCRIPTION
**Pull request by a-lyupa**
If there is partially downloaded local copy of repo,
continue downloading instead of creating new local
copy of the same repo.

Signed-off-by: Anastasia Lyupa <a.lyupa@samsung.com>

### CODE REVIEW

**Review by rayslava — APPROVED** [review]

File: combirepo/directory_downloader.py [hunk 1]
```diff
@@ -207,12 +207,11 @@ def mkdir():
                         with open(target, 'wb') as file_target:
                             file_target.write(contents)
     else:
-        if not os.path.isfile(target):
-            with sizes_lock:
-                sizes[target] = response.info().getheaders("Content-Length")[0]
-                logging.debug(
-                    "Setting size of {0} to {1}\n".format(target, sizes[target]))
-            download_file(response, target)
+        with sizes_lock:
```
Comment (ilya-palachev) [inline combirepo/directory_downloader.py:210]: And why is this needed? Won't we download the file twice?

Comment (rayslava) [reply combirepo/directory_downloader.py:210]: This prevents race condition for multi-process downloads reported earlier.

Comment (ilya-palachev) [reply combirepo/directory_downloader.py:210]: No-no, I'm not about `with sizes_lock:`, but about removing condition `if not os.path.isfile(target):`, which is done in the current patch. (Sorry for non-clear location of comment.)

File: combirepo/directory_downloader.py [hunk 2]
```diff
@@ -207,12 +207,11 @@ def mkdir():
                         with open(target, 'wb') as file_target:
                             file_target.write(contents)
     else:
-        if not os.path.isfile(target):
```
Comment (ilya-palachev) [inline combirepo/directory_downloader.py:210]: I have meant this line instead.

**Review by ilya-palachev — CHANGES_REQUESTED** [review]

File: combirepo/directory_downloader.py (hunk 1 above)
Comment (a-lyupa) [reply combirepo/directory_downloader.py:210]: There is the same check inside download_file(). And without removing the condition Downloading counter doesn't work.

Comment (ilya-palachev) [reply combirepo/directory_downloader.py:210]: OK, thanks! Now it's clear.

**Review by ilya-palachev — APPROVED** [review]

**Comment by ilya-palachev** [comment]
Good patch! I had also struggled from re-downloading of repositories that had happened very frequently.
